lib/google_scraping/dashboard/schemas/keyword_csv_file.ex

```diff
@@ -0,0 +1,78 @@
+defmodule GoogleScraping.Dashboard.Schemas.KeywordCSVFile do
+  use Ecto.Schema
+
+  import Ecto.Changeset
+  import GoogleScrapingWeb.Gettext
+
+  alias NimbleCSV.RFC4180, as: CSV
+
+  embedded_schema do
+    field :file, :map
+  end
+
+  @keywords_limit 1000
+  @keyword_min_length 1
+  @keyword_max_length 100
+
+  def keywords_limit, do: @keywords_limit
+  def keyword_min_length, do: @keyword_min_length
+  def keyword_max_length, do: @keyword_max_length
+
+  def create_changeset(keyword_file, attrs \\ %{}) do
+    keyword_file
+    |> cast(attrs, [:file])
+    |> validate_required([:file])
+    |> validate_file_type()
+  end
+
+  def parse(file_path) do
+    case get_keywords_list_from_file(file_path) do
+      {:ok, keywords_list} ->
+        case length(keywords_list) do
+          0 ->
+            {:error, :empty_file_error}
+
+          length when length > @keywords_limit ->
+            {:error, :file_is_too_long_error}
+
+          _ ->
+            {:ok, keywords_list}
+        end
```

Comment:
How about moving this p[art into a separate private function? so that `parse(file_path)` function becomes more readable. and can avoid nested `case` too. What do you think?
```suggestion
        some_private_function()
```